### PR TITLE
Enable yul optimizer by default.

### PR DESCRIPTION
**Pull request by ekpyron**

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -680,7 +681,8 @@ Allowed options)",
 			"Set for how many contract runs to optimize."
 			"Lower values will optimize more for initial deployment cost, higher values will optimize more for high-frequency usage."
 		)
-		(g_strOptimizeYul.c_str(), "Enable Yul optimizer in Solidity, mostly for ABIEncoderV2. Still considered experimental.")
+		(g_strOptimizeYul.c_str(), "Enable Yul optimizer in Solidity. Legacy option: the yul optimizer is enabled by default.")
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]:
```suggestion
		(g_strOptimizeYul.c_str(), "Enable Yul optimizer in Solidity. Legacy option: the yul optimizer is enabled as part of the general --optimize option.")
```

File: solc/CommandLineInterface.cpp [hunk 2]
```diff
@@ -948,6 +950,7 @@ bool CommandLineInterface::processInput()
 		Input inputLanguage = m_args.count(g_argYul) ? Input::Yul : (m_args.count(g_argStrictAssembly) ? Input::StrictAssembly : Input::Assembly);
 		Machine targetMachine = Machine::EVM;
 		bool optimize = m_args.count(g_argOptimize) || m_args.count(g_strOptimizeYul);
+		// TODO: should we exclude --no-optimize-yul here? Or enable optimizing by default?
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: In assembly mode, you should use ``--optimize`` instead of ``--optimize-yul``, so I think it can be ignored.

**Comment by chriseth** [comment]
Why draft?

**Comment by chriseth** [comment]
Weird failure.

**Comment by ekpyron** [comment]
I'm mainly waiting for the external tests to be fixed for this one - I'd say for this one they'd be really interesting... And I rebased earlier, the weird failure started then, I'll look into it!

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Tests are green. Will merge.

**Pull request merged**